src/main/java/module/TicTacToeService.java

```diff
@@ -0,0 +1,157 @@
+        ticTacToeBoard.setCol(col);
+        ticTacToeBoard.initboard(row, col);
+
+    }
+    void setBoardValue(Player player,int row, int col){
+        ticTacToeBoard.setBoard(row-1, col-1, player.getSymbol());
+    }
+    void printBoardValues(){
+        for(int i = 0; i < ticTacToeBoard.getRow(); i ++){
+            for(int j = 0; j < ticTacToeBoard.getCol(); j++){
+                System.out.print(ticTacToeBoard.getBoard(i, j) + " ");
+            }
+            System.out.println();
+        }
+    }
+    boolean colWiseCheck(){
+        int countValueX = 0;
+        int countValueO = 0;
+        for(int i = 0; i < ticTacToeBoard.getRow(); i++){
+            for(int j = 0; j < ticTacToeBoard.getCol(); j++){
+                if(ticTacToeBoard.getBoard(j,i) == 'X')
+                    countValueX++;
+                if(ticTacToeBoard.getBoard(j,i) == 'O')
+                    countValueO++;
+            }
+            if(countValueX == ticTacToeBoard.getCol())
+                return true;
+            else
+                countValueX = 0;
+            if(countValueO == ticTacToeBoard.getCol())
+                return true;
+            else
+                countValueO = 0;
+        }
+        return false;
+    }
+    boolean rowWiseCheck(){
+        int countValueX = 0;
+        int countValueO = 0;
+
+        for(int i = 0; i < ticTacToeBoard.getCol(); i++){
+            for(int j = 0; j < ticTacToeBoard.getRow(); j++){
+                if(ticTacToeBoard.getBoard(i,j) == 'X')
+                    countValueX++;
+                if(ticTacToeBoard.getBoard(i,j) == 'O')
+                    countValueO++;
+            }
+            if(countValueX == ticTacToeBoard.getCol())
+                return true;
+            else
+                countValueX = 0;
+            if(countValueO == ticTacToeBoard.getCol())
+                return true;
+            else
+                countValueO = 0;
+        }
+        return false;
+    }
+    boolean leftDiagonalCheck(){
+        int countValueX =0;
+        int countValueO = 0;
+        int leftDiagonal = 0;
+        for(int i = 0; i < ticTacToeBoard.getRow(); i++){
+            if(ticTacToeBoard.getBoard(i, leftDiagonal) == 'X')
+                countValueX++;
+            if(ticTacToeBoard.getBoard(i, leftDiagonal) == 'O')
+                countValueO ++;
+            leftDiagonal++;
+        }
+        if(countValueO == ticTacToeBoard.getCol() || countValueX == ticTacToeBoard.getCol())
+            return true;
+        return false;
+    }
+    boolean rightDiagonalCheck(){
+        int countValueX = 0;
+        int countValueO =0;
+        int rightDiaogal = ticTacToeBoard.getCol()-1;
+        for(int i = 0; i < ticTacToeBoard.getRow(); i++){
+            if(ticTacToeBoard.getBoard(i, rightDiaogal) == 'X')
+                countValueX++;
+            if(ticTacToeBoard.getBoard(i,rightDiaogal) == 'O')
+                countValueO ++;
+            rightDiaogal--;
```

Comment:
rightDiagonal is just ticTacToeBoard.getCol() - 1 - i.

Reply:
Value of "rowDiagonal--" and "getCol()-1-i" will be same after each iteration of i

Reply:
Yes, that's what I'm trying to say. A function which just receives row and col and can be used for all the validators. Your loop will contain 1-2 lines only if you do that.